src/System.Threading.Tasks.Channels/System/Runtime/CompilerServices/ValueAwaiter.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Provides an awaiter that can be created around an existing value or around
+    /// another async operation, represented by either a <see cref="Task{TResult}"/>
+    /// or an <see cref="IAwaiter{T}"/>. 
+    /// </summary>
+    /// <typeparam name="TResult">Specifies the type of the result.</typeparam>
+    [StructLayout(LayoutKind.Auto)]
+    public struct ValueAwaiter<TResult> : IAwaiter<TResult>
```

Comment:
Right now it's a place to stash it.  It could for example potentially move to System.Threading.Tasks.Extensions.dll (where ValueTask is).